HHS/Head-Start-TTADP
# [TTAHUB-945/954] Resolve create goals bugs reported by designers, rewrite file uploaders

## Description of change

This change addresses [some feedback provided by the design team](https://ocio-jira.acf.hhs.gov/browse/TTAHUB-945) on the _create goals form_.

Specifically:
- Draft goals are now included on the RTR table page 
- The sort is adjusted to show draft goals first
- Button order, spacing, and an icon were all fixed to match the mockups

A large part of this PR is built to address a bug with the objective file uploader/RTR, where a file couldn't be uploaded until the objective was save, creating a confusing user interaction. The file upload functionality was broken out into two components, one for objectives and one for activity reports, and a new user interaction created 

## How to test
- Check that the feedback mentioned in Jira 945 has been resolved

Upload file functionality still works
- On an activity report, you can upload a supporting attachment and it is saved properly, and can be removed
- On an activity report, you can upload a file to an objective and it is saved properly, and can be removed
- On the RTR, you can upload a file to an objective and it is saved properly, and can be removed

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-945
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-954

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
